### PR TITLE
Update CK code to ck_upstream of MIOpen

### PR DESCRIPTION
**Pull request by ltqin**
This PR  just update latest ck code   into MIOpen , reference https://github.com/ROCmSoftwarePlatform/composable_kernel/wiki/Sync-between-CK-and-MIOpen's-CK-subtree

### CODE REVIEW

**Comment by junliume** [comment]
Ping @atamazov @zjing14  and @asroy  for blessings of this PR.

**Comment by zjing14** [comment]
@ltqin I think the cppcheck warning is a false positive. I dropped the latest commit and suppressed the warning as below

`// cppcheck-suppress returnTempReference`

**Comment by atamazov** [comment]
@junliume Can you please indicate urgency/value of this PR?

**Comment by atamazov** [comment]
@ltqin Can you please elaborate PR description a bit more? For example, motivation (why the update is required), effects (what it improves or fixes), how it was tested etc. Understanding the goal helps to understand the ways of reaching it (i.e. to understand the code changes under review). Thanks.

**Comment by ltqin** [comment]
> @ltqin Can you please elaborate PR description a bit more? For example, motivation (why the update is required), effects (what it improves or fixes), how it was tested etc. Understanding the goal helps to understand the ways of reaching it (i.e. to understand the code changes under review). Thanks.

@atamazov This PR is just synchronizing CK code to miopen::ck_upstream branch. If needing a new CK solver, we can develop code on this basis, create a new PR and merge it to the "develop"  branch.

File: CMakeLists.txt [hunk 1]
```diff
@@ -25,6 +25,14 @@
 ################################################################################
 cmake_minimum_required( VERSION 3.5 )
 
+macro(set_var_to_condition var)
```
Comment (atamazov) [inline CMakeLists.txt:28]: Such changes should not appear in this PR. Merge develop into it.

**Review by atamazov — DISMISSED** [review]

**Comment by junliume** [comment]
Ping @zjing14 and @asroy for upstream blessings :)

**Comment by junliume** [comment]
@ltqin what is our plan for this PR? :) I recommend pining reviewers to avoid dragging it for too long.

**Comment by junliume** [comment]
> @ltqin what is our plan for this PR? :) I recommend pining reviewers to avoid dragging it for too long.

Ping @ltqin about the same question :) Happy New Year!

**Pull request closed without merging**